### DISCUSSION TITLE
In search result show/hide items based on approval status column/Filter items based on field choice.

### DISCUSSION THREAD

**Discussion by surfingknowledge** [microsoft-search/pnp-modern-search#2117] [Q&A] [answered]
Hi , Please I need help, 

We have a document library built upon approval process. So we would like to show only Approved documents/items in search results . I got 3 choices pending approved and rejected. It's okay for the doc to be shown in the doc library even if it's approved  b/s that's the backend but i want to hide all items not approved on my search page. 

Thank you in advance   , 

![image](https://user-images.githubusercontent.com/43178227/170729342-281cd3ad-a66c-4b79-b451-bbdadfa8c18a.png)

**Comment by FranckyC** [comment, marked as answer]
Hi @surfingknowledge, you can use a KQL condition with your column managed property name restricting only to "Approved" values. For that you must first map the crawled property associated to your SharePoint column to a "RefinableString" managed property in the search schema. Then use the condition like this in the query template option in Web Part properties:

`RefinableString01:Approved`

![image](https://user-images.githubusercontent.com/5659122/170781365-c5fb9712-8498-451f-a245-7699f9a6bc3d.png)

**Reply by surfingknowledge** [reply]
@FranckyC Thank you very much for the speedy response and support . Since i have my doc library path in the query template and didn't know how to mesh both , I used the Refinement filters box and it worked. Hmm I thought it couldn't be done but it turned out to be very simple solution. I see a very bright future and capability of this product. We immensely love it, It's expandable/scalable/ robust and has capacity to meet a broad business need /requirements. 

Appreciate the on going help. 

![image](https://user-images.githubusercontent.com/43178227/171680433-b5473e21-8c2e-47d4-a787-73f2574b8749.png)

**Reply by wobba** [reply]
You can move `RefinableString120:Approved` to the query template as KQL as well. Result is the same :)
